core/lib/dal/src/blocks_dal.rs

```diff
@@ -1272,12 +1272,20 @@ impl BlocksDal<'_, '_> {
     pub async fn get_ready_for_execute_l1_batches(
         &mut self,
         limit: usize,
+        commited_tx_confirmed: bool,
         max_l1_batch_timestamp_millis: Option<u64>,
     ) -> anyhow::Result<Vec<L1BatchWithMetadata>> {
         let raw_batches = match max_l1_batch_timestamp_millis {
             None => {
-                sqlx::query_as!(
-                    StorageL1Batch,
+                let (confirmed_at_not_null, join_on_eth_tx_history) = if commited_tx_confirmed {
+                    (
+                        "confirmed_at IS NOT NULL",
+                        "JOIN eth_txs_history ON eth_commit_tx_id = eth_tx_id",
```

Comment:
`eth_txs` table is not a part of this query so looks like it would also need to be joined?

Reply:
oh right missed this. I would probably just join `eth_txs` since it's the main table of eth sender. hisotry table is more of an implementation detail.

Of course this is more of an intuition, either tables would work